test/distributed/test_c10d.py

```diff
@@ -4082,6 +4082,54 @@ def allreduce_hook_gloo(
 
         self._run_and_verify_sparse_gradients(vanilla_model, ddp_model)
 
+    class AcceptsParam(torch.nn.Module):
+        def __init__(self, p, factor):
+            super().__init__()
+            self.a = p
+            self.f = factor
+
+        def forward(self, input):
+            return input + self.a * self.f
+
+    @requires_nccl()
+    def test_ddp_weight_sharing(self):
```

Comment:
would you please also add "@skip_if_not_multigpu"? otherwise it will fail when there is no gpus found, thanks

Reply:
done